src/python/pants/help/help_printer.py

```diff
@@ -198,6 +202,20 @@ def _print_all_tools(self) -> None:
         tool_help_cmd = f"{self._bin_name} help $tool"
         print(f"Use `{self.maybe_green(tool_help_cmd)}` to get help for a specific tool.\n")
 
+    def _print_all_api_types(self) -> None:
+        self._print_title("Plugin API Types")
+        api_type_descriptions: Dict[str, str] = {}
+        for api_type, rule_infos in self._all_help_info.rule_output_type_to_rule_infos.items():
```

Comment:
In your example the output is sorted alphabetically, but I'm not sure how?

Reply:
Perhaps we should omit types beginning with an underscore? Those are not intended to be used by general plugins.

Reply:
Theres a "sorted" call in the returned dict comprehension of get_rule_infos.

Reply:
Got it. So I suggest we filter out types that start with `_`, and then this is good to go.